src/viam/sdk/components/base/base.hpp

```diff
@@ -0,0 +1,86 @@
+/// @file components/base/base.hpp
+///
+/// @brief Defines a `Base` component.
+#pragma once
+
+#include <string>
+
+#include <viam/api/component/base/v1/base.pb.h>
+
+#include <viam/sdk/common/linear_algebra.hpp>
+#include <viam/sdk/common/proto_type.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource_manager.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup Base Classes related to the `Base` component.
+
+/// @class BaseSubtype
+/// @brief Defines a `ResourceSubtype` for the `Base` component.
+/// @ingroup Base
+class BaseSubtype : public ResourceSubtype {
+   public:
+    std::shared_ptr<ResourceServerBase> create_resource_server(
+        std::shared_ptr<ResourceManager> manager) override;
+    std::shared_ptr<ResourceBase> create_rpc_client(std::string name,
+                                                    std::shared_ptr<grpc::Channel> chan) override;
+    BaseSubtype(const google::protobuf::ServiceDescriptor* service_descriptor)
+        : ResourceSubtype(service_descriptor){};
+};
+
+/// @class Base base.hpp "components/base/base.hpp"
+/// @brief A `Base` is the platform that the other parts of a mobile robot attach to.
+/// @ingroup Base
+///
+/// This acts as an abstract parent class to be inherited from by any drivers representing
+/// specific base implementations. This class cannot be used on its own.
+class Base : public ComponentBase {
```

Comment:
@stuqdog - A naming thing. This line got my attention, because `Base` is overloaded here. When used as part of the name `ComponentBase` it being used to mean base class, I believe. But `Base` is a kind of component too, and generally a term of art in the Viam world. It feels muddled.

How would you feel about (in another review) removing `Base` from `ResourceServerBase`, `ComponentBase`, and `ResourceBase`, which I think are the only places it is currently used?

Reply:
That sounds reasonable to me, ticket [here](https://viam.atlassian.net/browse/RSDK-2971).